src/BuiltInTools/dotnet-watch/Build/ProjectInstanceId.cs

```diff
@@ -0,0 +1,6 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.DotNet.Watch;
+
+internal readonly record struct ProjectInstanceId(string projectPath, string targetFramework);
```

Comment:
Since this is used as a key in a dictionary and held in sets, does `ProjectInstanceId` need to define custom equality? E.g. should it compare project path with OrdinalIgnoreCase on Windows? Also, should these property names be pascal-cased?